javacord-api/src/main/java/org/javacord/api/entity/server/Server.java

```diff
@@ -2054,19 +2101,71 @@ default CompletableFuture<Void> banUser(String userId, int deleteMessageDays) {
      * @return A future to check if the ban was successful.
+     * @deprecated Use {@link #banUser(String, long, TimeUnit)} instead.
      */
+    @Deprecated
     default CompletableFuture<Void> banUser(long userId, int deleteMessageDays) {
         return banUser(Long.toUnsignedString(userId), deleteMessageDays);
     }
 
+    /**
+     * Bans the given user from the server.
+     *
+     * @param userId                The id of the user to ban.
+     * @param deleteMessageDuration The number of messages to delete within the duration.
+     *                              (Between 0 and 604800 seconds (7 days))
+     * @param unit                  The unit of time for the duration.
+     * @return A future to check if the ban was successful.
+     */
+    default CompletableFuture<Void> banUser(String userId, long deleteMessageDuration, TimeUnit unit) {
+        return banUser(userId, deleteMessageDuration, unit, null);
+    }
+
+    /**
+     * Bans the given user from the server.
+     *
+     * @param userId                The id of the user to ban.
+     * @param deleteMessageDuration The number of messages to delete within the duration.
+     *                              Between 0 and 604800 seconds (7 days))
+     * @param unit                  The unit of time for the duration.
+     * @return A future to check if the ban was successful.
+     */
+    default CompletableFuture<Void> banUser(long userId, long deleteMessageDuration, TimeUnit unit) {
+        return banUser(Long.toUnsignedString(userId), deleteMessageDuration, unit);
+    }
+
+    /**
+     * Bans the given user from the server.
+     *
+     * @param userId   The id of the user to ban.
+     * @param duration The number of messages to delete within the duration. (Between 0 and 604800 seconds (7 days))
+     * @return A future to check if the ban was successful.
+     */
+    default CompletableFuture<Void> banUser(String userId, Duration duration) {
+        return banUser(userId, duration, null);
+    }
+
+    /**
+     * Bans the given user from the server.
+     *
+     * @param userId   The id of the user to ban.
+     * @param duration The number of messages to delete within the duration. (Between 0 and 604800 seconds (7 days))
+     * @return A future to check if the ban was successful.
+     */
+    default CompletableFuture<Void> banUser(long userId, Duration duration) {
+        return banUser(Long.toUnsignedString(userId), duration);
+    }
+
     /**
      * Bans the given user from the server.
      *
      * @param userId            The id of the user to ban.
      * @param deleteMessageDays The number of days to delete messages for (0-7).
      * @param reason            The reason for the ban.
      * @return A future to check if the ban was successful.
+     * @deprecated Use {@link #banUser(String, long, TimeUnit)} instead.
      */
+    @Deprecated
     CompletableFuture<Void> banUser(String userId, int deleteMessageDays, String reason);
```

Comment:
```suggestion
    default CompletableFuture<Void> banUser(String userId, int deleteMessageDays, String reason) {
        return banUser(userId, deleteMessageDays, TimeUnit.DAYS, reason);
    }
```
With this we can reduce the implementation to just 1 method